src/main/java/org/betonquest/betonquest/objectives/ChestPutObjective.java

```diff
@@ -66,11 +66,10 @@ public ChestPutObjective(final Instruction instruction) throws InstructionParseE
         if (instruction.hasArgument("items-stay")) {
             chestTakeEvent = null;
         } else {
-            try {
-                chestTakeEvent = new ChestTakeEvent(new Instruction(instruction.getPackage(), new NoID(instruction.getPackage()), "chesttake " + location + " " + items));
-            } catch (final ObjectNotFoundException e) {
-                throw new InstructionParseException("Could not create inner chest take event: " + e.getMessage(), e);
-            }
+            final Instruction newInstruction = instruction.copy();
+            final CompoundLocation compoundLocation = newInstruction.getLocation();
+            final Instruction.Item[] instructionItemList = newInstruction.getItemList();
+            chestTakeEvent = new ChestTakeEvent(compoundLocation, instructionItemList);
```

Comment:
Can we not just use the existing `CompoundLocation loc` and call `getItemList` on the existing Instruction instead of coping it?

Reply:
no, because .getLocation() calls .getLocation(next()) but there is no next

Reply:
But you already extracted the location in line 57

Reply:
yes, but to get the item list, I need to call next() in any form and because i think this is more readable than gut calling next(), and now everything is compact